From 9dbf20f4d8a89f8904f7a4fa5d5dd5b1e34ff432 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Mon, 3 Nov 2025 11:09:11 +0900
Subject: [PATCH 53/54] fix lld exportSourceName

I've encountered a crash with combination of -Wl,--export-all-symbols, specifying a .def, linking an exe, and got ordinal too
large but failed reproduce with a simple test.
---
 lld/COFF/SymbolTable.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lld/COFF/SymbolTable.cpp b/lld/COFF/SymbolTable.cpp
index 26a2b4944dc6cb64718ac243a54daed6f95cfc24..8073d296ee8dd20b2c212bb441a422673b187411 100644
--- a/lld/COFF/SymbolTable.cpp
+++ b/lld/COFF/SymbolTable.cpp
@@ -1190,9 +1190,10 @@ static StringRef exportSourceName(ExportSource s) {
     return "/export";
   case ExportSource::ModuleDefinition:
     return "/def";
-  default:
-    llvm_unreachable("unknown ExportSource");
+  case ExportSource::Unset:
+    return "(unknown)";
   }
+  llvm_unreachable("unknown ExportSource");
 }
 
 // Performs error checking on all /export arguments.
-- 
2.52.0

